crates/ruff_linter/src/rules/flake8_bandit/rules/bad_file_permissions.rs

```diff
@@ -113,12 +111,12 @@ fn py_stat(call_path: &CallPath) -> Option<u16> {
     }
 }
 
-fn int_value(expr: &Expr, semantic: &SemanticModel) -> Option<u16> {
+fn int_value(expr: &Expr, semantic: &SemanticModel) -> Option<isize> {
     match expr {
         Expr::Constant(ast::ExprConstant {
-            value: Constant::Int(value),
+            value: Constant::Int(Int::Small(value)),
             ..
-        }) => value.to_u16(),
+        }) => Some(*value),
```

Comment:
So this truncates to `isize` (that is, we don't flag bad file permissions that exceed `isize`). Which seems not ideal. But it actually looks like an improvement, since before, we only flagged file permissions that could be cast to `u16`?

Reply:
Shouldn't all non `i64` values be reported as a bad file permissions? They're certainly wrong.

Reply:
I added a diagnostic for this.